packages/core/src/rules/common/hide-internals.ts

```diff
@@ -0,0 +1,65 @@
+import { Oas3Decorator, Oas2Decorator } from '../../visitors';
+import { isPlainObject } from '../../utils';
+
+const DEFAULT_HIDDEN_TAG = 'x-internal';
+const HIDDEN_TYPES = [
+  'PathItem',
+  'Operation',
+  'Schema',
+  'Response',
+  'RequestBody',
+  'Example',
+  'MediaType',
+  'Server',
+  'Callback',
+  'Parameter',
+];
+
+export const HideInternals: Oas3Decorator | Oas2Decorator = ({ hideTag }) => {
+  const hiddenTag = hideTag || DEFAULT_HIDDEN_TAG;
+  const isEmptyObject = (obj: any) => isPlainObject(obj) && Object.keys(obj).length === 0;
+  const isEmptyArray = (arr: any) => Array.isArray(arr) && arr.length === 0;
```

Comment:
add better typing instead of `any`

Reply:
@Marshevskyy I've moved these functions to utils module.
And about typing - the parameter is type `any` because we expected any type there and that's why we check: is it object or array. Also, we've already such js utils in [CLI](https://github.com/Redocly/openapi-cli/blob/master/packages/cli/src/js-utils.ts).